src/main/java/org/folio/search/service/setter/bibframe/LccnNormalizer.java

```diff
@@ -0,0 +1,64 @@
+package org.folio.search.service.setter.bibframe;
+
+import jakarta.validation.constraints.NotNull;
+import java.util.Optional;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+
+@Log4j2
+@Service
+public class LccnNormalizer {
```

Comment:
As this LCCN normalizer is used only by bibframe and in the module exists another logic related to LCCN normalization I suggest combining all related to LCCN logic in one place as having it separated may confuse.

Reply:
Hi @psmagin, @aleksei-pronichev 

This LccnNormalizer class is for normalizing Structure B LCCNs (Items acquired after 2001). I think the [normalizeLccn](https://github.com/folio-org/mod-search/blob/master/src/main/java/org/folio/search/utils/SearchUtils.java#L337) function in `SearchUtils` class is a generic normalizer that could be used for any kinds of LCCNs.

So, how about doing as follows?

Create an abstract class `LccnNormalizer` with two sub classes
1. `GenericLCCNNormalizer` -> Move the logic from [normalizeLccn](https://github.com/folio-org/mod-search/blob/master/src/main/java/org/folio/search/utils/SearchUtils.java#L337) function  to this class.
3. `StructureBLCCNormalizer` -> Rename this class to `StructureBLCCNormalizer`

Use `GenericLCCNNormalizer` in `AbstractLccnProcessor`
Use `StructureBLCCNormalizer` in `BibframeLccnProcessor`

Note: Documentation of StructureA and StructureB LCCNs - https://www.loc.gov/marc/lccn_structure.html#:~:text=On%20January%201%2C%202001%2C%20a,to%20make%20a%20LCCN%20unique.

Reply:
Done. Created new Classes and interfaces.
src/main/java/org/folio/search/service/lccn/LccnNormalizerStructureB.java
src/main/java/org/folio/search/service/lccn/LccnNormalizerStructureB.java
src/main/java/org/folio/search/service/lccn/DefaultLccnNormalizer.java